src/recipe/Baked.Recipe.Service.Application/Communication/Mock/MockCommunicationExtensions.cs

```diff
@@ -17,37 +18,48 @@ public static IClient<T> TheClient<T>(this Mocker mockMe,
         object? response = default,
         string? responseString = default,
         Exception? throws = default,
+        bool returnErrorResponse = false,
         List<object>? responses = default,
         bool? noResponse = default
     )
     {
         var mock = Moq.Mock.Get(mockMe.Spec.GiveMe.The<IClient<T>>());
 
-        var setup = () => mock.Setup(c => c.Send(It.Is<Request>(r =>
-            (url == default || r.UrlOrPath == url) &&
-            (path == default || r.UrlOrPath == path) &&
-            (urlEndsWith == default || r.UrlOrPath.EndsWith(urlEndsWith))
-        )));
+        var setup = () => mock.Setup(c =>
+            c.Send(It.Is<Request>(r =>
+                (url == default || r.UrlOrPath == url) &&
+                (path == default || r.UrlOrPath == path) &&
+                (urlEndsWith == default || r.UrlOrPath.EndsWith(urlEndsWith))
+            ),
+            It.IsAny<bool>()
+        ));
 
         if (throws is not null)
         {
-            setup().ThrowsAsync(throws);
+            if (returnErrorResponse)
+            {
+                setup().ReturnsAsync(new Response(throws is HandledException ? StatusCode.Handled : StatusCode.UnHandled, throws.Message ?? string.Empty));
```

Comment:
better to have `statusCode` directly from parameter instead of requiring both `throws` and `returnErrorResponse`

Reply:
you can still use existing `response` or `responseString` parameter values for content